manualdirectdebit.php

```diff
@@ -283,6 +279,10 @@ function manualdirectdebit_civicrm_custom($op, $groupID, $entityID, &$params) {
       $mandateDataGenerator = new CRM_ManualDirectDebit_Hook_Custom_DataGenerator($entityID, $params);
       $mandateDataGenerator->generateMandateData();
     }
+
+    $mandateStorageManager = new CRM_ManualDirectDebit_Common_MandateStorageManager();
+    $cancellationChecker = new CRM_ManualDirectDebit_Hook_Custom_CancellationBatchChecker($entityID, $params, $mandateStorageManager);
```

Comment:
@erawat many questions! The idea is to check if a mandate the was tied to a cancellations batch changed status, so I've moved this part to be in the check for edit or update operation. I can see it probably makes no sense running it on creation, for example.

I decided to instantiate MandateStorageManager here and inject into the class to avoid having hard dependencies in the code. This should make the new CancellationBatchChecker class easier to test.